astropy/table/table.py

```diff
@@ -335,6 +357,37 @@ def __get__(self, instance, owner_cls):
         return info
 
 
+class UnhideColumnsMethod:
```

Comment:
I was coming at this from the use case that both @Cadair and I have had, namely a Table subclass that has a known list of columns that are typically not interesting. This does beg the question of whether it should be allowed to hide columns that don't (yet) exist. That would allow putting a `self.hide_columns(not_interesting_columns)` into the subclass `__init__`. This seems like something that would be useful.

I see your point that the converse (HiddenColumns) could be useful as well. I need to think about this more to see if it is possible. The exact structure for `UnhideColumnsMethod` won't work but there might be a variation...

Worst case is being less fancy with:
```
t.hide_columns(cols)
t.unhide_columns(cols)
with t.hide_columns_context(cols):
     ...
with t.unhide_columns_context(cols):
    ...
```